libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -922,6 +1024,69 @@ string YulUtilFunctions::updateStorageValueFunction(Type const& _type, boost::op
 	});
 }
 
+string YulUtilFunctions::writeToMemoryFunction(Type const& _type)
+{
+	if (auto ref = dynamic_cast<ReferenceType const*>(&_type))
+	{
+		solAssert(
+			ref->location() == DataLocation::Memory,
+			"Can only update types with location memory."
+		);
+
+		return writeToMemoryFunction(*TypeProvider::uint256());
```

Comment:
I'm wondering whether it would be better to just do an mstore here, i.e. have an explicit function (so that the type is also visible in the name) and then use mstore without validation / cleanup.